chainable/__init__.py

```diff
@@ -0,0 +1,46 @@
+from functools import reduce, _initial_missing
+from typing import TypeVar, Callable, overload, Generic
+from collections.abc import Iterable, Iterator
+
+__all__ = ["Chainable"]
+
+T = TypeVar("T")
+S = TypeVar("S")
+
+def flat_map_helper(self: "Chainable[T]", mapper: Callable[[T], Iterable[S]]) -> Iterator[S]:
```

Comment:
We can use [PEP 563](https://peps.python.org/pep-0563/) (`from __future__ import annotations`) to not use quotes in this and other type annotations. I think it might be easier to maintain in the long term.